pkg/apis/batch/v1/types.go

```diff
@@ -0,0 +1,170 @@
+	ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty"`
+
+	// Selector is a label query over pods that should match the pod count.
+	// More info: http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors
+	Selector *LabelSelector `json:"selector,omitempty"`
+
+	// Template is the object that describes the pod that will be created when
+	// executing a job.
+	// More info: http://releases.k8s.io/HEAD/docs/user-guide/jobs.md
+	Template v1.PodTemplateSpec `json:"template"`
+}
+
+// JobStatus represents the current state of a Job.
+type JobStatus struct {
+
+	// Conditions represent the latest available observations of an object's current state.
+	// More info: http://releases.k8s.io/HEAD/docs/user-guide/jobs.md
+	Conditions []JobCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// StartTime represents time when the job was acknowledged by the Job Manager.
+	// It is not guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	StartTime *unversioned.Time `json:"startTime,omitempty"`
+
+	// CompletionTime represents time when the job was completed. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	CompletionTime *unversioned.Time `json:"completionTime,omitempty"`
+
+	// Active is the number of actively running pods.
+	Active int32 `json:"active,omitempty"`
+
+	// Succeeded is the number of pods which reached Phase Succeeded.
+	Succeeded int32 `json:"succeeded,omitempty"`
+
+	// Failed is the number of pods which reached Phase Failed.
+	Failed int32 `json:"failed,omitempty"`
+}
+
+type JobConditionType string
+
+// These are valid conditions of a job.
+const (
+	// JobComplete means the job has completed its execution.
+	JobComplete JobConditionType = "Complete"
+	// JobFailed means the job has failed its execution.
+	JobFailed JobConditionType = "Failed"
+)
+
+// JobCondition describes current state of a job.
+type JobCondition struct {
+	// Type of job condition, Complete or Failed.
+	Type JobConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	LastProbeTime unversioned.Time `json:"lastProbeTime,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime unversioned.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+}
+
+// A label selector is a label query over a set of resources. The result of matchLabels and
+// matchExpressions are ANDed. An empty label selector matches all objects. A null
+// label selector matches no objects.
+type LabelSelector struct {
```

Comment:
Yes-- there's already a generated release_1_2 client. Take a look at the
doc Chao is adding: #21436

On Wed, Feb 17, 2016 at 8:26 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/apis/batch/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/20551#discussion_r53270274>
> :
>
> > +	// Status of the condition, one of True, False, Unknown.
> > +	Status v1.ConditionStatus `json:"status"`
> > +	// Last time the condition was checked.
> > +	LastProbeTime unversioned.Time `json:"lastProbeTime,omitempty"`
> > +	// Last time the condition transit from one status to another.
> > +	LastTransitionTime unversioned.Time `json:"lastTransitionTime,omitempty"`
> > +	// (brief) reason for the condition's last transition.
> > +	Reason string `json:"reason,omitempty"`
> > +	// Human readable message indicating details about last transition.
> > +	Message string `json:"message,omitempty"`
> > +}
> > +
> > +// A label selector is a label query over a set of resources. The result of matchLabels and
> > +// matchExpressions are ANDed. An empty label selector matches all objects. A null
> > +// label selector matches no objects.
> > +type LabelSelector struct {
>
> Go packages and type names aren't part of the API. They are implementation
> details that are subject to change. That is also true for our "internal",
> unversioned client library, and any libraries in our main repo.
>
> @lavalamp <https://github.com/lavalamp> Will the generated versioned
> client be done in 1.2?
>
> LabelSelector is currently in unversioned.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20551/files#r53270274>.
>